GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -1223,35 +1223,14 @@ private HResult NotifyPreDeleteHandler(string virtualPath, bool isDirectory)
             bool isDirectory,
             ref NotificationType notificationMask)
         {
-            try
-            {
-                bool srcPathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(virtualPath);
-                bool dstPathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(destinationPath);
-
-                if (dstPathInDotGit)
-                {
-                    this.OnDotGitFileOrFolderChanged(destinationPath);
-                }
+            this.OnFileRenamed(virtualPath, destinationPath, isDirectory);
+        }
 
-                if (!(srcPathInDotGit && dstPathInDotGit))
-                {
-                    if (isDirectory)
-                    {
-                        this.FileSystemCallbacks.OnFolderRenamed(virtualPath, destinationPath);
-                    }
-                    else
-                    {
-                        this.FileSystemCallbacks.OnFileRenamed(virtualPath, destinationPath);
-                    }
-                }
-            }
-            catch (Exception e)
-            {
-                EventMetadata metadata = this.CreateEventMetadata(virtualPath, e);
-                metadata.Add("destinationPath", destinationPath);
-                metadata.Add("isDirectory", isDirectory);
-                this.LogUnhandledExceptionAndExit(nameof(this.NotifyFileRenamedHandler), metadata);
-            }
+        private void NotifyHardlinkCreated(
+            string relativePath,
+            string destinationPath)
+        {
+            this.OnHardLinkCreated(relativePath, destinationPath);

```

Comment:
Are these parameters ordered properly? The first parameter is ambiguous: `relativePath` could map to either `relativeTargetPath` or `relativeNewLinkPath`. But the second parameter sure looks like it maps better to `relativeTargetPath`, which makes me think these are swapped?

Reply:
These are ordered correctly.  

I agree the naming could be better (and I could change these names), but as a start I went with the approach of matching the parameter names declared in the ProjFS Managed API:

```
    /// <param name="relativePath">The path, relative to the virtualization root, of the file for
    ///     which the hard link was created.
    ///     <para>This parameter will be "" to indicate that the hard link name was created inside
    ///     the virtualization instance, i.e. a new hard link was created inside the virtualization
    ///     instance to a file that exists outside the virtualization instance.</para>
    /// </param>
    /// <param name="destinationPath">The path, relative to the virtualization root, of the new hard
    ///     link name.
    ///     <para>This parameter will be "" to indicate that the hard link name was created
    ///     outside the virtualization instance, i.e. a new hard link was created outside the
    ///     virtualization instance for a file that exists inside the virtualization instance.</para>
    /// </param>
```

Let me know if you think it's better to match the API names or use our own names.